src/bgw/job.c

```diff
@@ -246,9 +251,9 @@ ts_bgw_job_get_all(size_t alloc_size, MemoryContext mctx)
 	};
 	ScannerCtx scanctx = {
 		.table = catalog_get_table_id(catalog, BGW_JOB),
-		.index = InvalidOid,
+		.index = catalog_get_index(ts_catalog_get(), BGW_JOB, BGW_JOB_PKEY_IDX),
 		.data = &list_data,
-		.tuple_found = bgw_job_accum_tuple_found,
+		.tuple_found = bgw_job_accum_tuple_found_scheduled,
```

Comment:
I suggest adding a filter function  that does heap_getattr on scheduled instead of overloading the tuple found func.